FilePersistence/src/version_control/Commit.cpp

```diff
@@ -133,20 +133,22 @@ QStringList Commit::nodeLinesFromId(Model::NodeIdType id, bool findChildrenByPar
 	QStringList matches;
 	for (auto file : files())
 	{
-		int indexOfId = 0;
-		auto content = QString::fromUtf8(file->content());
-
-		indexOfId = content.indexOf(idText, indexOfId);
-		if (indexOfId != -1)
+		if (strstr(file->content(), idText.toLatin1().constData()))
 		{
-			int start, end;
-			if (isValidMatch(content, indexOfId, start, end, findChildrenByParentId))
+			int indexOfId = 0;
+			auto content = QString::fromUtf8(file->content());
```

Comment:
Here, instead of converting the entire file to QString, search for the previous '\n' (or the beginning of the file, whichever comes first) and the following '\n' (or the end of the file) and the only convert this line to UTF 8. You will need to modify the `isValidMatch` method to use this new interface.